### DISCUSSION TITLE
webstorm/Intellij typescript indexing is slow while Panda is compiling

### DISCUSSION THREAD

**Discussion by Yuripetusko** [chakra-ui/panda#1937] [Q&A]
Somewhat related to the other discussion about the slow compilation. My compilation is not as slow

```
🐼 info [hrtime] Extracted in (5.03ms)
- event compiled client and server successfully in 2.4s (6019 modules)
```

but what is very noticable is that when I change a single line of code and Panda does compilation, types hints and autocomplete doesn't work in Webstorm until the compilation finishes. Not sure if anything can be done. Also I noticed that it compiles even when I changed the code that is not really related to panda or styles, so maybe there's a way to adjust filtering of the watcher?

for the record I use postcss, ark and park-ui in nextjs

**Comment by astahmer** [comment]
hey ! saw your tweet first so I answered there but I'll answer here as well:

 iirc webstorm doesn't use tsc itself, maybe that can explain the slight differences with vscode (suggestions / unresponsiveness)
https://blog.jetbrains.com/webstorm/2023/12/try-the-future-typescript-engine-with-the-webstorm-next-program/

> Since introducing TypeScript support, WebStorm has operated with its own TypeScript engine. This engine has been effective in evaluating types, closely replicating the type evaluation mechanisms of the TypeScript compiler.

---
 
 -> Panda only re-generates the outdir (styled-system by default) files affecteds by a config change
e.g, update your "button" recipe and only the button files will be written again
otherwise, it only regenerate the CSS if you change your source files (config.include)

---

one thing that could help is to switch to the CLI instead of using the postcss plugin, as we can't control precisely when it will be re-run again as it mostly depends on the host bundler's implementation (vite / webpack (next) / etc)

whereas in the CLI we directly watch the filesystem, so it should only re-run when needed
